presto-main/src/test/java/com/facebook/presto/operator/TestGroupByHash.java

```diff
@@ -293,4 +299,90 @@ else if (type == BIGINT) {
         long[] expected = IntStream.range(0, actual.length).mapToLong(i -> entryBytes * expectedSize << (i + 1)).toArray();
         assertEquals(actual, expected);
     }
+
+    @Test(dataProvider = "dataType")
+    public void testPageYield(Type type)
+            throws Exception
+    {
+        // Create a page with positionCount >> expected size of groupByHash
+        int length = 1_000_000;
+        Block valuesBlock;
+        if (type == VARCHAR) {
+            valuesBlock = createStringSequenceBlock(0, length);
+        }
+        else if (type == BIGINT) {
+            valuesBlock = createLongSequenceBlock(0, length);
+        }
+        else {
+            throw new IllegalArgumentException("unsupported data type");
+        }
+        Block hashBlock = getHashBlock(ImmutableList.of(type), valuesBlock);
+        Page page = new Page(valuesBlock, hashBlock);
+        AtomicInteger counter = new AtomicInteger();
+        ExtraMemoryReservation forceYield = extraMemoryInBytes -> counter.incrementAndGet() % 3 == 0;
```

Comment:
Don't call this `forceYield`. Don't use count-based approach for this test.

Instead, use 2 AtomicInt, one for current requested, one for currently allowed. In the while loop below, increase the latter to be a reasonable value after every yield. Also verify that without increasing the allowed value, calling process again will result in immediate return.

Change log(length) to a literal number. Change `assertGreaterThanOrEqual` to `assertEquals`. Add comment to explain where the number come from. i.e. reduce the amount of computation in tests to bare minimum.